src/main/java/sparksoniq/jsoniq/item/Item.java

```diff
@@ -163,31 +159,57 @@ public static boolean checkEquality(Item v1, Item v2) {
         return compareItems(v1, v2) == 0;
     }
 
-    public abstract List<Item> getItems() throws OperationNotSupportedException;
+    public List<Item> getItems() {
+        throw new RuntimeException("Item is not an array or object.");
+    };
 
-    public abstract Item getItemAt(int i) throws OperationNotSupportedException;
+    public Item getItemAt(int i) {
+        throw new RuntimeException("Item is not an array.");
+    };
 
-    public abstract void putItem(Item value) throws OperationNotSupportedException;
+    public void putItem(Item value) {
+        throw new RuntimeException("Item is not an array.");
+    };
 
-    public abstract List<String> getKeys() throws OperationNotSupportedException;
+    public List<String> getKeys() {
+        throw new RuntimeException("Item is not an object.");
+    };
 
-    public abstract Collection<? extends Item> getValues() throws OperationNotSupportedException;
+    public Collection<? extends Item> getValues() {
+        throw new RuntimeException("Item is not an array or object.");
+    };
 
-    public abstract Item getItemByKey(String s) throws OperationNotSupportedException;
+    public Item getItemByKey(String s) {
+        throw new RuntimeException("Item is not an object.");
+    };
 
-    public abstract void putItemByKey(String s, Item value) throws OperationNotSupportedException;
+    public void putItemByKey(String s, Item value) {
+        throw new RuntimeException("Item is not an object.");
+    };
 
-    public abstract int getSize() throws OperationNotSupportedException;
+    public int getSize() {
+        throw new RuntimeException("Item is not an array.");
+    };
 
-    public abstract String getStringValue() throws OperationNotSupportedException;
+    public String getStringValue() {
+        throw new RuntimeException("getStringValue: Item is not a string.");
```

Comment:
Is there a reason why the exception message for getStringValue is different than other methods? I'd suggest keeping the standard similar to other methods.